common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -116,6 +116,13 @@ public Job updateJob(Job job) {
     public Resource getResourceForJob(String jobUuid, String fileName) throws MalformedURLException {
         Job job = getJobByJobUuid(jobUuid);
 
+        User user = userService.getCurrentUser();
+        if (!user.equals(job.getUser())) {
+            log.error("User attempted to download a file where they had a valid UUID, but was not logged in as the " +
+                    "user that created the job");
+            throw new InvalidJobAccessException("You don't have permissions to access the job " + jobUuid);
+        }
+
```

Comment:
I am _very_ happy we caught this.

Reply:
Let's move this check into `getJobByJobUuid`. That way we also enforce access checks in `getJobStatus` and `deleteRequest`.

Reply:
Let's also rename `getJobByJobUuid` into `getAuthorizedJobByJobUuid`